### PR TITLE
[exporter/clickhouse] Fix insert metrics with duplicate scope data

### LINKED ISSUE #21082: metrics scopeName error
### Component(s)

exporter/clickhouse

### What happened?

**Describe the bug**
 When I use a custom meter and set instrumentationScopeName，but the scopeName is not effective in the clickhouse.
i used java agent version is 1.24.0, collector version is v0.75.0

**Steps to reproduce**
image
image
image

image

### Collector version

v0.75.0

## Environment
OS: (e.g., "Ubuntu 16.04")

### OpenTelemetry Collector configuration

```yaml
extensions:
  health_check:
  pprof:
    endpoint: 0.0.0.0:1777
  zpages:
      endpoint: 0.0.0.0:55679
receivers:
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
      http:
        endpoint: 0.0.0.0:4318
  jaeger:
    protocols:
      grpc:
        endpoint: 0.0.0.0:14251
      thrift_compact:  
        endpoint: 0.0.0.0:6931
processors:
  batch:
exporters:
  jaeger:
    endpoint: xxx
    tls:
      insecure: true
  kafka/metrics: 
    brokers:
      - xxx
    protocol_version: 2.0.0
    topic: otel_metrics_test
    encoding: otlp_json
    producer: 
      compression: snappy
  kafka/logs: 
    brokers:
      - xxx
    protocol_version: 2.0.0
    topic: otel_logs_test
    encoding: otlp_json
    producer: 
      compression: snappy
  clickhouse: 
    endpoint: xxx
    ttl_days: 30
    timeout: 5s
    database: xxx
    connection_params: {"dial_timeout":"200ms","max_execution_time":60}
    username: xxx
    password: xxxx
    logs_table_name: otel_logs
    traces_table_name: otel_traces
    metrics_table_name: otel_metrics
    sending_queue: 
      queue_size: 5000
    retry_on_failure:
      enabled: true
      initial_interval: 5s
      max_interval: 30s
      max_elapsed_time: 300s
service:
  pipelines:
    traces:
      receivers: [otlp, jaeger]
      processors: [resource, batch]
      exporters: [jaeger, clickhouse]
    metrics:
      receivers: [otlp]
      processors: [resource, batch]
      exporters: [kafka/metrics, clickhouse]
    logs:
      receivers: [otlp]
      processors: [resource, batch]
      exporters: [kafka/logs, clickhouse]
  extensions: [health_check, pprof, zpages]
```

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Frapschen**
**Description:** <Describe what has changed.>

Fix insert metrics with duplicate scope data

**Link to tracking Issue:** 

resolve: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21082

**Testing:** <Describe what testing was performed and which tests were added.>

add a new UT test to show this bug is fixed

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/clickhouseexporter/exporter_metrics.go [hunk 1]
```diff
@@ -70,12 +70,12 @@ func (e *metricsExporter) shutdown(ctx context.Context) error {
...
 		res := metrics.Resource()
-		metaData.ResAttr = attributesToMap(res.Attributes())
-		metaData.ResURL = metrics.SchemaUrl()
 		for j := 0; j < metrics.ScopeMetrics().Len(); j++ {
+			metaData := internal.MetricsMetaData{}
+			metaData.ResAttr = attributesToMap(res.Attributes())
+			metaData.ResURL = metrics.SchemaUrl()
 			rs := metrics.ScopeMetrics().At(j).Metrics()
 			metaData.ScopeURL = metrics.ScopeMetrics().At(j).SchemaUrl()
 			metaData.ScopeInstr = metrics.ScopeMetrics().At(j).Scope()
```
Comment (dmitryax) [inline exporter/clickhouseexporter/exporter_metrics.go:81]: Maybe better to update `MetricsModel.Add` signature to receive resource and scope objects separately? Then you don't need to recreate metaData for resource on every iteration over scopes.

File: exporter/clickhouseexporter/internal/summary_metrics.go [hunk 2]
```diff
@@ -156,7 +157,7 @@ func (s *summaryMetrics) insert(ctx context.Context, db *sql.DB) error {
 	return nil
 }
 
-func (s *summaryMetrics) Add(metrics any, metaData *MetricsMetaData, name string, description string, unit string) error {
+func (s *summaryMetrics) Add(resAttr map[string]string, resURL string, scopeInstr pcommon.InstrumentationScope, scopeURL string, metrics any, name string, description string, unit string) error {
```
Comment (dmitryax) [inline exporter/clickhouseexporter/internal/summary_metrics.go:160]: What about making the signature shorter this way?

```suggestion
func (s *summaryMetrics) Add(res pmetric.ResourceMetrics, scope pmetric.ScopeMetrics, metrics any, name string, description string, unit string) error {
```

File: exporter/clickhouseexporter/internal/summary_metrics.go [hunk 3]
```diff
@@ -166,8 +167,13 @@ func (s *summaryMetrics) Add(metrics any, metaData *MetricsMetaData, name string
 		metricName:        name,
 		metricDescription: description,
 		metricUnit:        unit,
-		metadata:          metaData,
-		summary:           summary,
+		metadata: &MetricsMetaData{
+			ResAttr:    resAttr,
+			ResURL:     resURL,
+			ScopeURL:   scopeURL,
+			ScopeInstr: scopeInstr,
```
Comment (dmitryax) [inline exporter/clickhouseexporter/internal/summary_metrics.go:174]: Then you'll have
```suggestion
			ResAttr:    res.Resource().Attributes(),
			ResURL:     res.SchemaUrl(),
			ScopeURL:   scope.SchemaUrl(),
			ScopeInstr: scope.Scope(),
```

Comment (dmitryax) [reply exporter/clickhouseexporter/internal/summary_metrics.go:174]: nvm. you'd need to call `attributesToMap` on every scope

File: exporter/clickhouseexporter/internal/summary_metrics.go (hunk 2 above)
Comment (dmitryax) [reply exporter/clickhouseexporter/internal/summary_metrics.go:160]: Please ignore this comment

**Comment by Frapschen** [comment]
> Please fix linter errors

done

**Review by dmitryax — APPROVED** [review]

**Pull request merged**